examples/tests/valid/user-defined-resources-captures.w

```diff
@@ -0,0 +1,73 @@
+bring cloud;
+
+resource MyResource {
+  my_resource: cloud.Bucket;
+  my_str: str;
+  my_num: num;
+  my_bool: bool;
+  array_of_str: Array<str>;
+  map_of_num: Map<num>;
+  set_of_str: Set<str>;
+
+  array_of_queues: Array<cloud.Queue>;
```

Comment:
Do we also want to add a test for `Array<UserDefinedResource>`?
Also what about:
```wing
struct MyStruct {
  b: cloud.Bucket;
  n: num;
}
```
And testing `Array<MyStruct>`?

Reply:
We do however have struct literals, I think this should work:
```
array_of_structs: Array<cloud.QueueProps> [
  cloud.QueueProps { timeout: 60s, initialMessages: ["Chicks", "with", "bricks", "come"] },
  cloud.QueueProps { timeout: 120s, initialMessages: ["Chicks", "with", "blocks", "come"] }
];